scripts/publish/aws-s3.sh

```diff
@@ -22,16 +22,54 @@ set -e
 
 ./scripts/build/check-dependency.sh aws
 
-if [ "$#" -ne 1 ]; then
-  echo "Usage: $0 <file>" 1>&2
+function usage() {
+  echo "Usage: $0"
+  echo ""
+  echo "Options"
+  echo ""
+  echo "    -f <file>"
+  echo "    -b <s3 bucket>"
+  echo "    -v <version>"
+  echo "    -t <release type (production|snapshot)>"
   exit 1
+}
+
+ARGV_FILE=""
+ARGV_BUCKET=""
+ARGV_VERSION=""
+ARGV_RELEASE_TYPE=""
+
+while getopts ":f:b:v:t" option; do
+  case $option in
+    f) ARGV_FILE="$OPTARG" ;;
+    b) ARGV_BUCKET="$OPTARG" ;;
+    v) ARGV_VERSION="$OPTARG" ;;
+    t) ARGV_RELEASE_TYPE="$OPTARG" ;;
+    *) usage ;;
+  esac
+done
+
+if [ -z "$ARGV_FILE" ] || \
+   [ -z "$ARGV_BUCKET" ] || \
+   [ -z "$ARGV_VERSION" ] || \
+   [ -z "$ARGV_RELEASE_TYPE" ]
+then
+  usage
 fi
 
-ETCHER_VERSION=`node -e "console.log(require('./package.json').version)"`
-S3_BUCKET="resin-production-downloads"
+FILENAME=$(basename "$ARGV_FILE")
+
+if [ "$ARGV_RELEASE_TYPE" == "production" ]; then
+  ROOT_KEY=etcher
```

Comment:
Done! I'm now accepting a "product name" option here. Let me know if you think we can rename this to something better!